### PR TITLE
[MRG+1] [py3] replace rfc822 with email.utils

### PR DESCRIPTION
**Pull request by nyov**
@kmike, @dangra 
nice sprint :)
i'm probably already too late with this

### CODE REVIEW

File: scrapy/downloadermiddlewares/robotstxt.py [hunk 1]
```diff
@@ -65,5 +65,9 @@ def _logerror(self, failure, request, spider):
 
     def _parse_robots(self, response):
         rp = robotparser.RobotFileParser(response.url)
-        rp.parse(response.body.splitlines())
+        # FIXME: unsure if we should do this here
+        # (or enforce the .encoding attr in `Response`)
+        if not hasattr(response, 'encoding'):
+            response.encoding = 'utf-8'
```
Comment (kmike) [inline scrapy/downloadermiddlewares/robotstxt.py]: Adding an extra fallback here sounds OK to me, for the case server returns some strange headers for /robots.txt request, but the file contents is ok. But it is better not to patch the response. It should use `response.body_as_unicode()` when possible, and fall back to `response.body.decode('...')` when it fails.

**Comment by kmike** [comment]
Looks good, thanks!

**Comment by kmike** [comment]
+1 to merge it once response patching is removed

**Comment by nyov** [comment]
Aye, gotcha. `body_as_unicode()` had slipped my memory.
But I split that commit from this tree, I rewrote the tests too much, and put it here #1391 for some feedback.

**Comment by kmike** [comment]
LGTM. I'll leave it to @dangra to merge because he was working on porting cookie code to Python 3.

**Comment by nyov** [comment]
Just to make sure, but you probably noticed, I put this on top of the other `tmp-py3` branch. It likely fails the py3 tests on master.
If I should remove the cookie commit again, let me know.

**Comment by kmike** [comment]
I'm brave enough to merge this PR by myself if there are no cookie changes (I've checked the related modules and tests, and they look fine after your changes). Daniel said he was working on cookies, so it is better to wait for him if the PR contains PY3 cookie changes.

**Comment by nyov** [comment]
Sure, I'll just stash that commit back to my other pile of bitrot for now :)

**Comment by kmike** [comment]
Thanks @nyov!

**Comment by dangra** [comment]
:+1:

**Comment by nyov** [comment]
And I dropped some more in #1393 :)

**Pull request merged**